src/test/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/BalanceSheetServiceImplTests.java

```diff
@@ -87,341 +114,619 @@
+        balanceSheet.getCalledUpShareCapitalNotPaid().setPreviousAmount((long)1000);
+        return balanceSheet;
+    }
 
+    private CompanyAccountApi createFirstYearCompanyAccountsObject() {
         CompanyAccountApi companyAccounts = new CompanyAccountApi();
-        companyAccounts.setNextAccounts(nextAccounts);
-        companyAccounts.setLastAccounts(lastAccounts);
-
-        CompanyProfileApi companyProfile = new CompanyProfileApi();
-        companyProfile.setAccounts(companyAccounts);
-
-        String currentPeriodHeading = "currentPeriodHeading";
-
-        when(accountsDatesHelper.generateBalanceSheetHeading(
-                any(LocalDate.class), any(LocalDate.class), anyBoolean())).thenReturn(currentPeriodHeading);
+        companyAccounts.setNextAccounts(createCurrentPeriodObject());
+        return companyAccounts;
+    }
 
-        when(accountsDatesHelper.convertDateToLocalDate(any(Date.class))).thenReturn(LocalDate.now());
+    private CompanyAccountApi createMultipleYearCompanyAccountsObject() {
+        CompanyAccountApi companyAccounts = createFirstYearCompanyAccountsObject();
+        companyAccounts.setLastAccounts(createPreviousPeriodObject());
+        return companyAccounts;
+    }
 
-        BalanceSheetHeadings balanceSheetHeadings = balanceSheetService.getBalanceSheetHeadings(companyProfile);
+    private LastAccountsApi createPreviousPeriodObject() {
+        DateTime previousPeriodEnd = new DateTime("2018-01-01");
+        DateTime previousPeriodStart = new DateTime("2017-01-01");
 
-        assertEquals(currentPeriodHeading, balanceSheetHeadings.getCurrentPeriodHeading());
+        LastAccountsApi lastAccounts = new LastAccountsApi();
+        lastAccounts.setPeriodStartOn(previousPeriodStart);
+        lastAccounts.setPeriodEndOn(previousPeriodEnd);
+        return lastAccounts;
     }
 
-    @Test
-    @DisplayName("Get Balance Sheet Headings when Previous Period exists and period end date is not null")
-    void getBalanceSheetHeadingsPreviousPeriodNotNull() {
-
+    private NextAccountsApi createCurrentPeriodObject() {
         DateTime currentPeriodStart = new DateTime("2018-01-01");
-        DateTime currentPeriodEnd = new DateTime("2018-01-01");
-        DateTime previousPeriodEnd = new DateTime("2018-01-01");
+        DateTime currentPeriodEnd = new DateTime("2019-01-01");
 
         NextAccountsApi nextAccounts = new NextAccountsApi();
         nextAccounts.setPeriodStartOn(currentPeriodStart);
         nextAccounts.setPeriodEndOn(currentPeriodEnd);
+        return nextAccounts;
+    }
 
-        LastAccountsApi lastAccounts = new LastAccountsApi();
-        lastAccounts.setPeriodEndOn(previousPeriodEnd);
-
-        CompanyAccountApi companyAccounts = new CompanyAccountApi();
-        companyAccounts.setNextAccounts(nextAccounts);
-        companyAccounts.setLastAccounts(lastAccounts);
+    private void mockApiClientGet() {
```

Comment:
This method may be better renamed as `mockApiClientGetSmallFull` for clarity

Reply:
I'm not sure about that as we are in the small full parent package and that would mean I would have to add small full onto every method and test to keep consistency for that same rule. Welcome to be overruled if others feel the same but I do feel that because we are in the small full package that it is to be assumed that it will only be doing small full related functionality